variables.tf

```diff
@@ -124,8 +124,8 @@ variable "availability_zone_count" {
   description = "Number of Availability Zones for the domain to use."
 
   validation {
-    condition     = contains([2, 3], var.availability_zone_count)
-    error_message = "The availibility zone count must be 2 or 3."
+    condition     = contains([2, 3], var.availability_zone_count) && var.zone_awareness_enabled == "true"
```

Comment:
**INVALID TERRAFORM**: Variable validation [conditions can refer only to the containing variable](https://developer.hashicorp.com/terraform/language/expressions/custom-conditions#input-variable-validation).